embassy-nrf/src/pwm.rs

```diff
@@ -136,12 +128,28 @@ impl<'d, T: Instance> SequencePwm<'d, T> {
         })
     }
 
-    /// Start or restart playback
+    /// Start or restart playback. Takes at least one sequence along with its
+    /// configuration. Optionally takes a second sequence and/or its configuration.
+    /// In the case where no second sequence is provided then the first sequence
+    /// is used. In the case where no second sequence configuration is supplied,
+    /// the first sequence configuration is used. The sequence mode applies to both
+    /// sequences combined as one.
     #[inline(always)]
-    pub fn start(&mut self, sequence: &'d [u16], times: SequenceMode) -> Result<(), Error> {
-        slice_in_ram_or(sequence, Error::DMABufferNotInDataMemory)?;
-
-        if sequence.len() > 32767 {
+    pub fn start(
+        &mut self,
+        sequence0: &'d [u16],
+        sequence_config0: SequenceConfig,
+        sequence1: Option<&'d [u16]>,
```

Comment:
It's a bit strange that you can pass `sequence1=Some, sequence_config1=None` or vice versa, shouldn't it be all or nothing? Perhaps move the buffer into SequenceConfig, maybe rename it to Sequence, then take `Option<Sequence<'d>>`?

Reply:
Done as of commit 12ce02457438c1dd1566f3f3c43537ae4f54f699